src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -50,11 +49,14 @@ void detach() {
 
     /**
      * Emit the specified {@link ProviderEvent}.
-     * 
+     *
      * @param event   The event type
      * @param details The details of the event
      */
     public void emit(ProviderEvent event, ProviderEventDetails details) {
```

Comment:
I think this is a good point, and we can probably reduce this visibility without considering this a breaking change since events are still labeled "experimental" so that would break our stability guarantee.

Reply:
@chrfwow did this but I had to revert it. We had multiple providers in the contribs who used an encapsulation strategy and used this publicly instead of in a subclass.